lib/cli/test/unit/Cardano/CLISpec.hs

```diff
@@ -46,9 +57,54 @@ import qualified Data.Text.IO as TIO
 spec :: Spec
 spec = do
     describe "Can perform roundtrip textual encoding & decoding" $ do
+        textRoundtrip $ Proxy @Iso8601Time
         textRoundtrip $ Proxy @(Port "test")
         textRoundtrip $ Proxy @MnemonicSize
 
+    describe "Can decode valid ISO 8601 strings" $ do
+        describe "UTC+0 timezone (Z)" $ do
+            canDecodeValidIso8601Time "2008-09-15T15:53:00Z"
+            canDecodeValidIso8601Time "2008-09-15T15:53:00.1Z"
+            canDecodeValidIso8601Time "2008-09-15T15:53:00.12Z"
+        describe "UTC+0 timezone (manually specified)" $ do
+            canDecodeValidIso8601Time "2008-09-15T15:53:00+00:00"
+            canDecodeValidIso8601Time "2008-09-15T15:53:00.1+00:00"
+            canDecodeValidIso8601Time "2008-09-15T15:53:00.12+00:00"
+        describe "UTC+8 timezone (manually specified)" $ do
+            canDecodeValidIso8601Time "2008-09-15T15:53:00+08:00"
+            canDecodeValidIso8601Time "2008-09-15T15:53:00.1+08:00"
+            canDecodeValidIso8601Time "2008-09-15T15:53:00.12+08:00"
+        describe "UTC-8 timezone (manually specified)" $ do
+            canDecodeValidIso8601Time "2008-09-15T15:53:00-08:00"
+            canDecodeValidIso8601Time "2008-09-15T15:53:00.1-08:00"
+            canDecodeValidIso8601Time "2008-09-15T15:53:00.12-08:00"
+
```

Comment:
It seems we don't support the basic format though. On slack you wrote "Or should we allow both Basic and Extended?" — is this intended?

If intended I hope people will not be confused by the API and CLI requiring different date-formats. I imagine a developer integrating with the API, that wants to pull values from a JSON log and try some requests with the CLI. Then the dates cannot simply be copy-pasted. But maybe that's a rare scenario, I have no idea.

Otherwise, I'd suggest at least one negative test to make it clear to the reader that we don't the support them:

Quick suggestion:

```suggestion

    describe "Cannot decode basic format ISO 8601 strings" $ do
        describe "UTC+0 timezone (Z)" $ do
            cannotDecodeInvalidIso8601Time "20190712T133821Z"
            cannotDecodeInvalidIso8601Time "20080915T1553001Z"
            cannotDecodeInvalidIso8601Time "20080915T15530012Z"
            
```

Reply:
> If intended I hope people will not be confused by the API and CLI requiring different date-formats. I imagine a developer integrating with the API, that wants to pull values from a JSON log and try some requests with the CLI. Then the dates cannot simply be copy-pasted.

I do agree. We should actually support the basic format in the CLI. We don't support the extended format in the API because it can be confusing, but it would be wrong to not support both in the CLI.

Reply:
> I do agree. We should actually support the basic format in the CLI. We don't support the extended format in the API because it can be confusing, but it would be wrong to not support both in the CLI.

I agree that's a very sensible suggestion.

I've added support for ISO 8601 basic in addition to ISO 8601 extended. Because the `iso8601-time` package doesn't support ISO 8601 basic, I've removed that package and used the parsing functions from `Data.Time.Format` to add the necessary support. The `FromText` instance now decodes both basic and extended strings.

Reply:
@Anviking wrote:
> If intended I hope people will not be confused by the API and CLI requiring different date-formats. I imagine a developer integrating with the API, that wants to pull values from a JSON log and try some requests with the CLI. Then the dates cannot simply be copy-pasted. But maybe that's a rare scenario, I have no idea.

I agree that we should support both. I've added support for the basic format, along with a set of test to ensure that valid strings parse (and that various kinds of invalid strings do not). Thanks for highlighting this! :)